src/imex.rs

```diff
@@ -472,7 +473,54 @@ async fn import_backup(context: &Context, backup_to_import: impl AsRef<Path>) ->
             .query_row(
                 "SELECT file_name, file_content FROM backup_blobs WHERE id = ?",
                 paramsv![file_id],
-                |row| Ok((row.get::<_, String>(0)?, row.get::<_, Vec<u8>>(1)?)),
+                // cs: This is the affected line NOW for producing the error. But
+                // here get is awaiting for String or another type to come back?
+                // |row| Ok((row.get::<_, String>(0)?, row.get::<_, Vec<u8>>(1)?)),
+
+                // cs: this is the secure replacement code for |row| ...
+                |row| {
+                    // Statement  'let name: String = row.get(0)?;'  is an assumption!
+                    //
+                    // This assumption that data matches String type is only true with clean utf8!
+                    // This simple "row.get(0) only works when TEXT column is ok and String conversion
+                    // from clean utf8 encoding is possible.
+                    // Every bad utf8 encoded data in TEXT column crashes this assumption and
+                    // switches to Err(_) path.
+                    let name: String = match row.get(0) {
+                        Ok(name) => name, // good utf8 filename
+                        Err(_) => {
+                            // bad utf8 encoding in filename
+                            match row.get_raw(0) {
+                                ValueRef::Null => "Null Value".to_string(),
+                                ValueRef::Text(t) => {
+                                    // column type in db is TEXT
+                                    let name = String::from_utf8_lossy(t).to_string();
+                                    info!(
```

Comment:
Shouldn't this be a `warn!`?

Reply:
Yes at first, but I think as column type of database is ok in that case and only filename needs to be converted an `info!` would be more appropriate, isn't it?
All other cases are unusual or issues and should be `warn!`

Reply:
This is the important code point I mention in #1902